woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php

```diff
@@ -377,48 +266,23 @@ public function enqueue_scripts() {
 	}
 
 	/**
-	 * Add the "One-click Purchase" button below "Add Cart" on Product Page
-	 */
-	public function add_button() {
-		global $product;
-
-		if ($this->gateway === false) {
-			return;
-		}
-
-		if (
-			$product->product_type == 'external' ||
-			!$this->customer_can() ||
-			!$this->gateway->is_available() ||
-			$this->gateway->configs->settings['one_click'] !== 'yes'
-		) {
-			return;
-		}
-
-		if ( $product->product_type == 'variable' ) {
-			add_action( 'woocommerce_after_single_variation', array( $this, 'print_button' ) );
-		} else {
-			add_action( 'woocommerce_after_add_to_cart_button', array( $this, 'print_button' ) );
-		}
-	}
-
-	/**
 	 * Check if the custom has all required data required by EBANX
 	 *
 	 * @return boolean If the user has all required data, return true
 	 */
 	protected function customer_has_ebanx_required_data() {
 		$card = current( array_filter( (array) array_filter( get_user_meta( $this->userId, '_ebanx_credit_card_token', true ) ), function ( $card ) {
-					return $card->token == WC_EBANX_Request::read('_ebanx_one_click_token');
-				} ) );
+			return $card->token == WC_EBANX_Request::read('ebanx-one-click-token');
+		} ) );
+
 		$names = $this->gateway->names;
 
 		$_POST['ebanx_token'] = $card->token;
 		$_POST['ebanx_masked_card_number'] = $card->masked_number;
 		$_POST['ebanx_brand'] = $card->brand;
-		$_POST['ebanx_billing_cvv'] = WC_EBANX_Request::read('_ebanx_one_click_cvv');
+		$_POST['ebanx_billing_cvv'] = WC_EBANX_Request::read('ebanx-one-click-cvv');
 		$_POST['ebanx_is_one_click'] = true;
-		$_POST['ebanx_billing_instalments'] = WC_EBANX_Request::read('_ebanx_one_click_installments');
+		$_POST['ebanx_billing_instalments'] = WC_EBANX_Request::read('ebanx-credit-card-installments');
 
 		$_POST[$names['ebanx_billing_brazil_document']] = get_user_meta( $this->userId, '_ebanx_billing_brazil_document', true );
```

Comment:
We're not using superglobals anymore, please replace `$_POST[<key>] = <value>` with `empty(WC_EBANX_Request::set(<key>, <value>))`